### PR TITLE
CAKE-4662 | :sparkles: Add useInerval hook and Countdown component

### PR DESCRIPTION
**Pull request by vforge**
- [x] Create `hooks/useInterval`
- [x] Document `hooks/useInterval`
- [x] Create `components/Countdown`
- [x] Test `components/Countdown`
- [x] Create `components/StyledCountdown`

### CODE REVIEW

File: source/components/Countdown/index.js [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+    const [value, setValue] = useState(START);
+
+    useInterval(() => {
+        const newValue = value - 1;
+        setValue(newValue);
+        onTick(newValue);
+    }, value > 0 ? ONE_SECOND : null);
+
+    // eslint-disable-next-line
+    const rangeClassName = (value < LOW ? 'is-low' : (value < HIGH ? 'is-medium' : 'is-high'));
```
Comment (jamieshepherd) [inline source/components/Countdown/index.js]:
```suggestion
    const rangeClassName = (value <= LOW ? 'is-low' : (value <= HIGH ? 'is-medium' : 'is-high'));
```

**Comment by jamieshepherd** [comment]
I'm not sure if the Countdown component here should actually be responsible for the timer - it might be more useful if it's just the UI representation of the current time passed as a prop - because the Quiz question itself has to also run its own timer (to control other UI elements).

**Comment by vforge** [comment]
@jamieshepherd which ones? I don't see this component being useful without the timer

**Comment by jamieshepherd** [comment]
I'm saying the reverse should be the case, the countdown shouldn't be responsible for the timer - the question should be. Countdown should just be the reflection of the current time

**Comment by mklucsarits** [comment]
I disagree, I think having the timing and UI together in one component is a good idea. This repo doesn't need to have only UI-focussed components.

**Comment by jamieshepherd** [comment]
Just to be clear this is what you are suggesting

Screen Shot 2019-04-05 at 16 06 00

That seems unnecessary to me 🤷‍♂️

File: config/jest.setup.js [hunk 2]
```diff
@@ -1,7 +1,12 @@
...
 import Enzyme from 'enzyme'; // eslint-disable-line import/no-extraneous-dependencies
+import { ThemeProvider } from 'styled-components';
+
+import theme from '../source/theme';
 
 Enzyme.configure({ adapter: new Adapter() });
 
 global.mockComponent = name => props => <mocked-component name={name} {...props} />;
+
+global.mountWithThemeProvider = child => Enzyme.mount(<ThemeProvider theme={theme}>{child}</ThemeProvider>);
```
Comment (JoshuaRogan) [inline config/jest.setup.js]: Good idea 👍

File: source/hooks/useInterval/index.js [hunk 3]
```diff
@@ -0,0 +1,22 @@
+import { useEffect, useRef } from 'react';
+
+const useInterval = (callback, delay) => {
```
Comment (JoshuaRogan) [inline source/hooks/useInterval/index.js]: What do you think about linking something like https://overreacted.io/making-setinterval-declarative-with-react-hooks/ or any other source explaining why this is needed. It's not obvious at first.

Comment (vforge) [reply source/hooks/useInterval/index.js]: 👍

**Pull request merged**